created check on bid.order for validating items

EagleEye25 opened the pull request.

cis-shubham-t committed:
- created the mock-service for the market.service

cis-shubham-t committed:
- mock responses file which contains the all kind of response json with the respective params and cmd(s).

cis-shubham-t committed:
- Use the mock.market.service instead of market service.

cis-shubham-t committed:
- move the get profile id related stuff in to the loadsubmitter() method.

cis-shubham-t committed:
- Added the Sleep class in the util.

cis-shubham-t committed:
- Update the proposal mock response file.
- Write load proposal method test case
-

cis-shubham-t committed:
- Added added the mock response for the vote(optionId), vote get, vote result data in mock json file.
- Created the vote related mock services.

cis-shubham-t committed:
- Added the sleep methods for seconds and ms seconds.

cis-shubham-t committed:
- Proposal service test cases.
- Update proposal mock response data.

Commit:
created check on bid.order for validating items

Commit:
fix lint

Commit:
minor change

Commit:
Update bid.service.ts

cis-shubham-t committed:
- Update the test cases.
- Move the proposal service mock data in the _test directory.
- Update the market mock service.

cis-shubham-t committed:
run test

zaSmilingIdiot committed:
Merge pull request #1325 from cis-shubham-t/tc-proposal-service

proposal service test cases

Commit:
Merge branch 'market-beta' of https://github.com/particl/particl-desktop into checkout-imporvement-expiredItems

zaSmilingIdiot committed:
Update bid.service.ts

Corrects reference to expired time value on the listing lookup.

Commit:
Merge branch 'market' of https://github.com/particl/particl-desktop into checkout-imporvement-expiredItems

zaSmilingIdiot committed:
Update bid.service.ts

slight perf adjustment

Commit:
Update bid.service.ts

Commit:
Merge branch 'checkout-imporvement-expiredItems' of https://github.com/EagleEye25/partgui into checkout-imporvement-expiredItems

zaSmilingIdiot commented:
Just commenting here to indicate the reason it has not yet been merged:
The current change returns the user to step 1 in the checkout process if a listing item expires while on step 3 (the final step). The user is then prompted to remove the expired item(s) from their cart in order to continue. If other items existed in their cart, then when navigating to step 2 after removing the expired item(s), the address inputs are now blank (although if a pre-saved address was selected, this dropdown is still selected). 

This is not an issue with this code change specifically, but it does need to be resolved when merging this change in in order to prevent the problem from occurring.
The fix for this scenario is being worked on at the moment.

Commit:
imporvements

Commit:
Merge branch 'checkout-imporvement-expiredItems' of https://github.com/EagleEye25/partgui into checkout-imporvement-expiredItems

Commit:
Removes unnecessary fixes. Moves the address update/creation on the checkout process to occur after the bid has been placed (bid uses correct address anyway).

Commit:
Backing out of current changes due to incorrect merged branches.
Revert "Removes unnecessary fixes. Moves the address update/creation on the checkout process to occur after the bid has been placed (bid uses correct address anyway)."

This reverts commit 8448f977e351aed66472b740985aa168760f5a55.

Commit:
Removes unnecessary fixes. Moves the address update/creation on the checkout process to occur after the bid has been placed (bid uses correct address anyway).

Commit:
Revert "Merge branch 'market' of https://github.com/particl/particl-desktop into checkout-imporvement-expiredItems"

This reverts commit 24160eba2769df911f9e06545d2052e48f76473e, reversing
changes made to 445ba10f8cf3b3648d6bb45c63bdca7337995771.

Commit:
reduce the time allocation used to determine listing expiry in the cart before doing the posting (5s for first item and 2.5s for every additional item).
